lisa/runners/lisa_runner.py

```diff
@@ -616,12 +620,15 @@ def _attach_failed_environment_to_result(
         # so deployment failure can be tracked.
         environment.platform = self.platform
         result.environment = environment
+
+        result.subscribe_log(self._log)
```

Comment:
Does it need to subscribe runner here? The below line `result.handle_exception` should have enough log. The runner is called in multi threads, and may bring logs from other cases.

Reply:
You are right, logs in deploy_environment should be sufficient.

Reply:
subscribe here is too late, it won't capture the log generated when deploying.

Reply:
Are you recommending to subscribe and unsubscribe in _deploy_environment_task itself rather than _attach_failed_environment_to_result ?

Reply:
Yes, subscribe in deploy and prepare, not in the attach failed.

Reply:
I am trying to find which test result to use in _prepare_environment

Reply:
You can manually create an exception in prepare, so it can be tested easier.

Reply:
As discussed offline, prepare has a different way of selecting test result. In this PR we focus on improving Deployment logging alone

Reply:
The subscribe should include deploy step as well. If only subscribe here, it looks only the exception is included, no other logs during deploying.

Reply:
Sorry, I didn't realize that I did not push this change. 
Pushed now